sspl_test/framework/utils/iem.py

```diff
@@ -0,0 +1,108 @@
+        "KAFKA_ACTIVE" : ["0050020002", "kafka"]
+    }
+
+    # EVENT_STRING = { event_code : [description, impact, recommendation] }
+    EVENT_STRING = {
+        "0050010001" : ["ipmitool command execution error.",
+            "Server resource monitoring through IPMI halted.",
+            "Reinstall/reconfigure ipmitool package."],
+        "0050010002" : ["ipmitool command execution success again.",
+            "Server resource monitoring through IPMI enabled again.",
+            ""],
+        "0050010003" : ["hdparm command execution error.",
+            "Server local drives monitoring through hdparm halted.",
+            "Reinstall/reconfigure hdparm package."],
+        "0050010004" : ["hdparm command execution success again.",
+            "Server local drives monitoring through hdparm enabled again.",
+            ""],
+        "0050010005" : ["smartctl command execution error.",
+            "Unable to fetch server drive SMART test results and related health info.",
+            "Reinstall/reconfigure smartmonotools package."],
+        "0050010006" : ["smartctl command execution success again.",
+            "Enabled again to fetch server drive SMART test results and related health info.",
+            ""],
+        "0050010007" : ["udisks2 is not installed.",
+            "Unable to fetch server drive info using systemd dbus interface.",
+            "Reinstall/reconfigure udisks2 package."],
+        "0050010008" : ["udisks2 is available.",
+            "Enabled again to fetch server drive info using systemd dbus interface.",
+            ""],
+        "0050020001" : ["Kafka service is not in active state.",
+            "Cortx health alerts may not be delivered to consumers like CSM.",
+            "Reconfigure/start kafka service."],
+        "0050020002" : ["Kafka service is back in active state.",
+            "Cortx health alerts will get delivered to consumers like CSM.",
+            ""]
+    }
+
+    def create_iem_fields(self, event, severity, event_type=None):
+        event_code = event[0]
+        event_name = event[1]
+        description = self.EVENT_STRING[event_code][0]
+        self.generate_iem(event_name, event_code, severity, description)
+
+    def iem_fault(self, event):
+        event = self.EVENT_CODE[event]
+        severity = self.Severity["ERROR"]
+        self.create_iem_fields(event, severity)
+
+    def iem_fault_resolved(self, fault_res_event):
+        severity = self.Severity["INFO"]
+        event = self.EVENT_CODE[fault_res_event]
+        event_type = "fault_resolved"
+        self.create_iem_fields(event, severity, event_type)
+
+    @staticmethod
+    def generate_iem(module, event_code, severity, description):
+        """Generate iem and send it to a MessgaeBroker."""
+        try:
+            logger.info(f"Sending IEM alert for module:{module}"
+                        f" and event_code:{event_code}")
+            EventMessage.send(module=module, event_id=event_code,
```

Comment:
so we are restricting test till IEM send, which is fine too. Any further issue falls under IEM framework scope, where send, subscription & receive test flow should ideally be there. @sachinpunadikar  hope this unit test is covered in IEM framework tests already

Reply:
in the iem test, we are sending the mock iem alert and testing alert is received or not in the specific format